storage/innobase/trx/trx0trx.cc

```diff
@@ -544,6 +545,7 @@ void trx_disconnect_prepared(trx_t *trx)
   trx->read_view.close();
   trx_sys.trx_list.freeze();
   trx->is_recovered= true;
+  trx->need_wait_report= false;
```

Comment:
Zero initialization should be guaranteed by a debug assertion that you should add to `trx_create()`.